File: 패스트캠퍼스/코테 1/2.py

def solution():

    N = 4
    fry = [2, 2, 1, 3]
    clean = [2, 4, 3, 2]
    chickens = 2
    time = []
    start = 0
    answer = 100000000
    for i in range(len(fry)):
        time.append(fry[i] + clean[i])
    end = max(time) * chickens

    while start <= end:
        mid = (start + end) // 2
        sum = 0
        for i in time:
            sum += mid // i

        if sum > chickens:
            end = mid - 1
        elif sum < chickens:
            start = mid + 1
        elif sum == chickens:
            end -= 1
            answer = min(answer, mid)

    print(answer)


solution()

File: 패스트캠퍼스/자체코테 7번째/2.js
function solution(times) {
  const answer = 0;
  const arr = Array.from({ length: 100000 }, () => 0);
  for (let i = 0; i < times.length; i++) {
    for (let j = times[i][0]; j < times[i][1]; j++) {
      arr[j] += 1;
    }
  }

  return Math.max(...arr);
}

console.log(solution([[14, 18], [12, 15], [15, 20], [20, 30], [5, 14]]));
